### PR TITLE
Fix BaseFilterSet not using multiple choice filters

### PR DESCRIPTION
**Pull request by timizuoebideri1**
# Closes: #3214 
# What's Changed

- Override `BaseFilterSet.filter_for_lookup` to add logic that checks if the lookup type is exact and if the field has attribute `choices` (as only IntegerField/CharField with choices can have this attribute), and returns MulipleChoiceFilter if the conditions are meet. 
- I also removed this block of code `elif existing filter.extra.get("choices")` because we now expect the choice field to be `MulipleChoiceFilter`, which would require all relevant lookup expressions of a `MulipleChoiceFilter` to be included.

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Review by jathanism — APPROVED** [review]

File: nautobot/extras/tests/test_filters.py
Comment (bryanculver) [inline nautobot/extras/tests/test_filters.py:1088]: With all of these string to list changes I feel like this is going to be a breaking change, if not at least for all apps who's tests were expecting to filter on a choice of one.

**Review by bryanculver — CHANGES_REQUESTED** [review]

File: nautobot/extras/tests/test_filters.py
Comment (glennmatthews) [reply nautobot/extras/tests/test_filters.py:1088]: Yeah, it's only a breaking change to things that are programmatically using the filterset (UI and API filtering should "just work") but that's definitely worth calling out.

Comment (jathanism) [reply nautobot/extras/tests/test_filters.py:1088]: A bug fix that is a breaking change! What is the world coming to!

Comment (timizuoebideri1) [reply nautobot/extras/tests/test_filters.py:1088]: I guess illd have to rebase to next?

File: nautobot/utilities/filters.py
Comment (glennmatthews) [inline nautobot/utilities/filters.py:13]: I'd prefer to skip this additional import and just reference it as `django_filters.MultipleChoiceFilter` where needed, personally.

File: nautobot/utilities/filters.py
Comment (glennmatthews) [inline nautobot/utilities/filters.py:727]: Couldn't this be:

```suggestion
        if lookup_type == "exact" and getattr(field, "choices", None):
            return MultipleChoiceFilter, {"choices": field.choices}
        return super().filter_for_lookup(field, lookup_type)
```

File: nautobot/extras/tests/test_filters.py
Comment (bryanculver) [reply nautobot/extras/tests/test_filters.py:1088]: Since this isn't a ground-breaking bug (I'm not seeing hard-stop issues for users) can I request this go into `next` so as to not induce unnecessary churn for apps and their automated tests?

File: nautobot/utilities/tests/test_filters.py
Comment (glennmatthews) [inline nautobot/utilities/tests/test_filters.py:665]: Am I missing something, or is there no test case for the existing `"choicefield"` filter?

File: nautobot/extras/tests/test_filters.py
Comment (glennmatthews) [reply nautobot/extras/tests/test_filters.py:1088]: I'd be fine with keeping this in `develop` as long as we add a clear note in the release-notes.

File: nautobot/utilities/tests/test_filters.py
Comment (timizuoebideri1) [reply nautobot/utilities/tests/test_filters.py:665]: Nope, or not that I could find

Comment (glennmatthews) [reply nautobot/utilities/tests/test_filters.py:665]: Would you mind adding one then? :-)

Comment (timizuoebideri1) [inline nautobot/utilities/tests/test_filters.py:665]:
```suggestion
        for field in ["multiplechoicefield", "charchoicefield", "choicefield"]:
```

**Review by bryanculver — APPROVED** [review]

**Review by glennmatthews — COMMENTED** [review]
Does this have any impact on the use of these filters in the UI, or how they are presented in the advanced filtering form?

**Review by glennmatthews — CHANGES_REQUESTED** [review]
Would like to clarify the impact of this on the UI and forms before it gets merged.

**Comment by jathanism** [comment]
> Would like to clarify the impact of this on the UI and forms before it gets merged.

With screenshots? :)

**Comment by timizuoebideri1** [comment]
It does not seem to have any impact on UI filtering both on Default and Advance

Screenshot 2023-02-27 at 23 21 39
Screenshot 2023-02-27 at 23 21 45
Screenshot 2023-02-27 at 23 21 49
Screenshot 2023-02-27 at 23 21 57
Screenshot 2023-02-27 at 23 23 07

File: nautobot/dcim/tests/test_filters.py
Comment (glennmatthews) [inline nautobot/dcim/tests/test_filters.py:1090]: Should be able to change most of these affected tests to entries in `generic_filter_test` rather than separate test functions now that these are multi-value filters, though it may take a bit of modifying to the test data to make it diverse enough.

**Review by glennmatthews — APPROVED** [review]

File: nautobot/dcim/tests/test_filters.py
Comment (timizuoebideri1) [reply nautobot/dcim/tests/test_filters.py:1090]: It does not need to be done on this PR right?

Comment (glennmatthews) [reply nautobot/dcim/tests/test_filters.py:1090]: I won't block this PR on waiting for this change but it should be a pretty straightforward one to do. Fine with me if you want to open a new issue or add a comment on #3213 calling out these tests.

**Pull request merged**
